### PR TITLE
refactor(ui): rewrite form fields

### PR DESCRIPTION
**Pull request by g1eny0ung**
Signed-off-by: Yue Yang <g1enyy0ung@gmail.com>

#### Changed

- The theme palette has been updated because of the new form fields.
- Extracts old `Scope` component to `components/` for next use.

#### Fixed

- Package `@ui/mui-extends` should use `peerDependencies` instead of `dependencies`.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: ui/app/src/@types/mui.d.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 Chaos Mesh Authors.
```
Comment (fewdan) [inline ui/app/src/@types/mui.d.ts:2]: s/2021/2022/g

File: ui/app/src/components/FormField/AutocompleteField.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Chaos Mesh Authors.
```
Comment (fewdan) [inline ui/app/src/components/FormField/AutocompleteField.tsx:2]: ditto

File: ui/app/src/@types/mui.d.ts (hunk 1 above)
Comment (g1eny0ung) [reply ui/app/src/@types/mui.d.ts]: Thx! I've already corrected both of them. 😃

**Review by fewdan — APPROVED** [review]
LGTM

File: ui/packages/mui-extends/.storybook/preview.js [hunk 3]
```diff
@@ -6,4 +9,13 @@ export const parameters = {
       date: /Date$/,
     },
   },
+  layout: 'centered',
 }
+
+export const decorators = [
+  (Story) => (
+    <ThemeProvider>
```
[the remark this answers is not in the document]
Comment (g1eny0ung) [reply ui/packages/mui-extends/.storybook/preview.js:17]: ignore

File: ui/packages/mui-extends/.storybook/preview.js [hunk 4]
```diff
@@ -6,4 +9,13 @@ export const parameters = {
...
+      <Story />
+    </ThemeProvider>
```
[the remark this answers is not in the document]
Comment (g1eny0ung) [reply ui/packages/mui-extends/.storybook/preview.js:19]: ignore

**Review by STRRL — APPROVED** [review]
LGTM! 🤩

**Comment by STRRL** [comment]
"GitHub Packages" status goes to "degrade", which blocks our e2e test.

I would retry after it recovery.

**Pull request merged**